File: app/src/main/java/com/lpc/test/utils/LogUtil.java
package com.lpc.test.utils;

import android.util.Log;


/**
 * 功能:
 * <p>
 * 描述: 正式Tag为class名称，lipc0113临时测试log
 * <p>
 * Created by lipc0113 on 2017/12/21.
 */

public class LogUtil {

    private static final String TAG = "lipc0113";

    private LogUtil() {
        throw new UnsupportedOperationException("u can't instantiate me...");
    }

    /**
     * show error note
     *
     * @param error
     */
    public static void e(String error) {

        Log.e(TAG, "---> " + error);
    }


    public static void e(String tag, String error) {
        Log.e(tag, "---> " + error);
    }

    public static void e(Throwable throwable) {

        if (throwable != null) {

            final StackTraceElement[] stackTrace = throwable.getStackTrace();
            final String message = throwable.getMessage();
            StringBuffer sb = new StringBuffer();
            sb.append("fatal :");
            sb.append(message);
            sb.append("\n");
            for (int i = 0; i < stackTrace.length; i++) {
                sb.append(stackTrace[i]);
            }

            Log.e(TAG, "---> " + sb.toString());
        }
    }

    public static void e(String tag, Throwable ex) {

        if (ex != null) {

            final StackTraceElement[] stackTrace = ex.getStackTrace();
            final String message = ex.getMessage();
            StringBuffer sb = new StringBuffer();
            sb.append("fatal :");
            sb.append(message);
            sb.append("\n");
            for (int i = 0; i < stackTrace.length; i++) {
                sb.append(stackTrace[i]);
            }

            Log.e(tag, "---> " + sb.toString());
        }
    }

    /**
     * show debug note
     *
     * @param infor
     */
    public static void d(String infor) {

        Log.d(TAG, "---> " + infor);
    }

    public static void d(String tag, String infor) {
        Log.d(tag, "---> " + infor);
    }

    public static void i(String infor) {

        Log.i(TAG, "---> " + infor);
    }

    public static void i(String tag, String infor) {
        Log.i(tag, "---> " + infor);
    }
}

File: kotlin-lib/src/main/java/com/lpc/test/kotlin_lib/utils/LogUtil.kt
package com.lpc.test.kotlin_lib.utils

import android.util.Log

/**
 * 功能:
 *
 *
 * 描述: 正式Tag为class名称，lipc0113临时测试log
 *
 *
 * Created by lipc0113 on 2017/12/21.
 */
class LogUtil private constructor() {
    companion object {
        private const val TAG = "lipc0113"

        /**
         * show error note
         *
         * @param error
         */
        fun e(error: String) {
            Log.e(TAG, "---> $error")
        }

        fun e(tag: String?, error: String) {
            Log.e(tag, "---> $error")
        }

        fun e(throwable: Throwable?) {
            if (throwable != null) {
                val stackTrace = throwable.stackTrace
                val message = throwable.message
                val sb = StringBuffer()
                sb.append("fatal :")
                sb.append(message)
                sb.append("\n")
                for (i in stackTrace.indices) {
                    sb.append(stackTrace[i])
                }
                Log.e(TAG, "---> $sb")
            }
        }

        fun e(tag: String?, ex: Throwable?) {
            if (ex != null) {
                val stackTrace = ex.stackTrace
                val message = ex.message
                val sb = StringBuffer()
                sb.append("fatal :")
                sb.append(message)
                sb.append("\n")
                for (i in stackTrace.indices) {
                    sb.append(stackTrace[i])
                }
                Log.e(tag, "---> $sb")
            }
        }

        /**
         * show debug note
         *
         * @param infor
         */
        fun d(infor: String) {
            Log.d(TAG, "---> $infor")
        }

        fun d(tag: String?, infor: String) {
            Log.d(tag, "---> $infor")
        }

        fun i(infor: String) {
            Log.i(TAG, "---> $infor")
        }

        fun i(tag: String?, infor: String) {
            Log.i(tag, "---> $infor")
        }
    }

    init {
        throw UnsupportedOperationException("u can't instantiate me...")
    }
}